### PR TITLE
Add custom detection example

### PR DESCRIPTION
**Pull request by kiryldz**

### CODE REVIEW

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            // we need to lock pixels  explicitly because we will draw in another (main) thread
+            originalImage.lockPixels()
+            // associate underlying native ByteBuffer with our buffer
+            originalImage.copyPixels(buffer)
+            buffer.rewind()
+            // copy ByteBuffer to bitmap
+            frame.copyPixelsFromBuffer(buffer)
+            // now we will draw current detections on canvas
+            val canvas = Canvas(frame)
+            val paint = Paint(Paint.ANTI_ALIAS_FLAG)
```
Comment (alexander-kulikovskii) [inline Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: please, create paint when create such activity

File: Examples/src/main/res/layout/activity_custom_detection.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <ImageView
+        android:id="@+id/detections_view"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:scaleType="centerCrop"/>
+
+</FrameLayout>
```
Comment (alexander-kulikovskii) [inline Examples/src/main/res/layout/activity_custom_detection.xml]:
```suggestion
    <ImageView
       xmlns:android="http://schemas.android.com/apk/res/android"
        android:id="@+id/detections_view"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:scaleType="centerCrop"/>
```

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            val frame = Bitmap.createBitmap(
+                originalImage.size.imageWidth,
+                originalImage.size.imageHeight,
+                Bitmap.Config.ARGB_8888
+            )
+            // prepare direct ByteBuffer that will hold camera frame data
+            val buffer = ByteBuffer.allocateDirect(originalImage.sizeInBytes())
+            // we need to lock pixels  explicitly because we will draw in another (main) thread
+            originalImage.lockPixels()
+            // associate underlying native ByteBuffer with our buffer
```
Comment (yunikkk) [inline Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: `associate` is confusing me here. IMO the method call itself is clear enough, just copy pixels to the buffer

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            originalImage.copyPixels(buffer)
```
Comment (yunikkk) [inline Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: Seems pixels can be released right here, after we copied them to the buffer?

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+    // VisionEventsListener handles events from Vision SDK on background thread.
+    private val visionEventsListener = object : VisionEventsListener {
+
+        override fun onAuthorizationStatusUpdated(authorizationStatus: AuthorizationStatus) {}
+
+        override fun onFrameSegmentationUpdated(frameSegmentation: FrameSegmentation) {}
+
+        override fun onFrameDetectionsUpdated(frameDetections: FrameDetections) {
+            // prepare camera frame first thing
```
Comment (yunikkk) [inline Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: Seems if you split this callback into functions, e.g. smth like
```
fun convertImageToBitmap(image): Bitmap {...}
fun drawDetections(bitmap, detections) {...}
fun drawSingleDetection(canvas, detection) {...}
```
most comments will become unnecessary and code will be more readable, wdyt?

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt (hunk 4 above)
Comment (yunikkk) [reply Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: And if you extract image -> bitmap convertion code then it'll become even more clear

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt (hunk 3 above)
Comment (kiryldz) [reply Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: I overthought a bit.
Do not need `lockPixels()`/`unlockPixels()` at all here.

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt (hunk 4 above)
Comment (kiryldz) [reply Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: Yes, you're right.

File: Examples/src/main/res/layout/activity_custom_detection.xml (hunk 2 above)
Comment (kiryldz) [reply Examples/src/main/res/layout/activity_custom_detection.xml]: OK

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt (hunk 1 above)
Comment (kiryldz) [reply Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: Alright.

File: Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt (hunk 5 above)
Comment (kiryldz) [reply Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt]: Will do

File: Examples/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -66,10 +63,21 @@
...
+            android:label="@string/custom_detections_example"
+            android:screenOrientation="landscape">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
 
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+        <activity android:name=".CustomDetectionActivity"
+            android:screenOrientation="landscape" />
```
Comment (yunikkk) [inline Examples/src/main/AndroidManifest.xml]: No label?

Comment (kiryldz) [reply Examples/src/main/AndroidManifest.xml]: We have lo labels for Java activities.

**Pull request merged**